src/amberscript/parser.cc

```diff
@@ -2907,23 +2907,26 @@ Result Parser::ParseValues(const std::string& name,
     }
 
     if (type::Type::IsFloat(segs[seg_idx].GetFormatMode())) {
-      if (!token->IsInteger() && !token->IsDouble()) {
+      if (!token->IsInteger() && !token->IsDouble() && !token->IsHex()) {
         return Result(std::string("Invalid value provided to ") + name +
                       " command: " + token->ToOriginalString());
       }
 
+      double val = token->IsHex() ? static_cast<double>(token->AsHex())
+                                  : token->AsDouble();
       Result r = token->ConvertToDouble();
```

Comment:
Does this need to happen before the ->AsDouble in the line above?

Reply:
Actually this line is clearly useless. ConvertToDouble already handles hex.